src/pedersen.rs

```diff
@@ -371,22 +394,51 @@ mod tests {
     #[test]
     fn hash() {
         // Test vector from https://github.com/starkware-libs/crypto-cpp/blob/master/src/starkware/crypto/pedersen_hash_test.cc
-        let a = Fp::from_str_vartime(
-            "1740729136829561885683894917751815192814966525555656371386868611731128807883",
+        let a = StarkHash::from_hex_str(
+            "0x3d937c035c878245caf64531a5756109c53068da139362728feb561405371cb",
```

Comment:
Endianess changes around these parts make me uneasy together with `#[PrimeFieldReprEndianness = "big"]` change.

Is there a case asserting that parsing the previous decimal string == from_hex_str output? This could be just the old `Fp::from_str_vartime` with `StarkHash::from` conversion right, and latter comparison done on Fp's.

This way you could again grep for constants across implementations.

Reply:
The original Starkware test cases are actually specified in hex -- our previous tests had just hard converted them to decimal using some calculator I presume. This was required because the derived `Fp::from_str_vartime` only allowed for decimal strings.

> Is there a case asserting that parsing the previous decimal string == from_hex_str output? This could be just the old Fp::from_str_vartime with StarkHash::from conversion right, and latter comparison done on Fp's.

The [hash_to_field_element](https://github.com/eqlabs/pathfinder/pull/46/files#diff-f0eee2a44c5a3c97da1e7a426ec20ca56de887c914c77dc96dfb78839ff30d7bR416) and [field_element_to_hash](https://github.com/eqlabs/pathfinder/pull/46/files#diff-f0eee2a44c5a3c97da1e7a426ec20ca56de887c914c77dc96dfb78839ff30d7bR431) do test the different conversions against each other.

> Endianess changes around these parts make me uneasy together with #[PrimeFieldReprEndianness = "big"] change.
Yeah I checked with Michael that the internal endianess doesn't impact the result. So we can also trivially change back to little endian again with no impact so long as we maintain consistency.

I primarily changed to big endian because it seems to be the Ethereum standard. And BigInts are generally stored as BE so that they can be compared more easily (but that doesn't really apply here). It does mean that one can compare python BigInt bytes etc.